internal/ethapi/transaction_options.go

```diff
@@ -0,0 +1,89 @@
+package ethapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/state"
+)
+
+type AccountStorage struct {
+	RootHash  *common.Hash
+	SlotValue map[common.Hash]common.Hash
+}
+
+func (r *AccountStorage) UnmarshalJSON(data []byte) error {
+	var hash common.Hash
+	if err := json.Unmarshal(data, &hash); err == nil {
+		r.RootHash = &hash
+		return nil
+	}
+	return json.Unmarshal(data, &r.SlotValue)
+}
+
+func (r AccountStorage) MarshalJSON() ([]byte, error) {
+	if r.RootHash != nil {
+		return json.Marshal(*r.RootHash)
+	}
+	return json.Marshal(r.SlotValue)
+}
+
+type TransactionOpts struct {
+	KnownAccounts  map[common.Address]AccountStorage `json:"knownAccounts"`
+	BlockNumberMin *hexutil.Uint64                   `json:"blockNumberMin,omitempty"`
+	BlockNumberMax *hexutil.Uint64                   `json:"blockNumberMax,omitempty"`
+	TimestampMin   *hexutil.Uint64                   `json:"timestampMin,omitempty"`
+	TimestampMax   *hexutil.Uint64                   `json:"timestampMax,omitempty"`
+}
+
+func (o *TransactionOpts) Check(blockNumber uint64, timeStamp uint64, statedb *state.StateDB) error {
+	if o.BlockNumberMin != nil && blockNumber < uint64(*o.BlockNumberMin) {
+		return errors.New("BlockNumberMin condition not met")
+	}
+	if o.BlockNumberMax != nil && blockNumber > uint64(*o.BlockNumberMax) {
+		return errors.New("BlockNumberMax condition not met")
+	}
+	if o.TimestampMin != nil && timeStamp < uint64(*o.TimestampMin) {
+		return errors.New("TimestampMin condition not met")
+	}
+	if o.TimestampMax != nil && timeStamp > uint64(*o.TimestampMax) {
+		return errors.New("TimestampMax condition not met")
+	}
+	counter := 0
+	for _, account := range o.KnownAccounts {
+		if account.RootHash != nil {
+			counter += 1
+		} else if account.SlotValue != nil {
+			counter += len(account.SlotValue)
```

Comment:
I thought the limit was on the number of known accounts, not on the number of storage slots.